src/components/UI/ViewEditName.tsx

```diff
@@ -109,12 +136,13 @@ class ViewAndEditName extends React.Component {
 
   handleCancel = () => {
     this.toggleEditMode(false, {
-      value: this.props.name,
+      value: this.props.value,
       errorMessage: '',
+      editing: false,
     });
   };
 
-  handleChange = (e) => {
+  handleChange: React.ChangeEventHandler<ElementRef<'input'>> = (e) => {
```

Comment:
The event is actually attached to the input element, not to the ref. Same for the others that use a similar type.

```suggestion
  handleChange: React.ChangeEventHandler<HTMLInputElement>> = (e) => {
```